src/plugins/perception/tabletop-objects/tabletop_objects_thread.h

```diff
@@ -64,44 +64,63 @@ class TabletopVisualizationThreadBase;
 #endif
 
 /** @class OldCentroid "tabletop_objects_thread.h"
- * This class is used to save old centroids in order to check for reappearance
+ * This class is used to save old centroids in order to check for reappearance.
  */
 class OldCentroid {
 public:
-  /**
-   * Constructor
-   * @param id The ID which the centroid was assigned to
-   * @param centroid The position of the centroid
-   */
-  OldCentroid(const unsigned int &id, const Eigen::Vector4f &centroid)
-  : id_(id), age_(0), centroid_(centroid) { }
-  /** Copy constructor
-   * @param other The other OldCentroid */
-  OldCentroid(const OldCentroid &other)
-  : id_(other.getId()), age_(other.getAge()), centroid_(other.getCentroid()) { }
-  virtual ~OldCentroid() { }
-  // any class with Eigen::Vector4f needs a custom new operator
+	/**
+	 * Constructor
+	 * @param id The ID which the centroid was assigned to
+	 * @param centroid The position of the centroid
+	 */
+	OldCentroid(const unsigned int &id, const Eigen::Vector4f &centroid)
+	: id_(id), age_(0), centroid_(centroid) {}
+
+	/** Copy constructor
+	 * @param other The other OldCentroid */
+	OldCentroid(const OldCentroid &other)
+	: id_(other.get_id()), age_(other.get_age()), centroid_(other.get_centroid()) { }
+
+	/** Destructor. */
+	virtual ~OldCentroid() { }
+
+	/** Assignment operator.
+	 * @param other The other OldCentroid
+	 * @return reference to this instance */
+	OldCentroid& operator=(const OldCentroid &other)
+	{
+		id_       = other.get_id();
+		age_      = other.get_age();
+		centroid_ = other.get_centroid();
+		return *this;
+	}
+
+	// any class with Eigen::Vector4f needs a custom new operator
```

Comment:
I'm not sure if this was on purpose, but most of the class is now indented differently. It's inconsistent with the rest of the file. The other changes (`getAge` -> `get_age`) are good.

Please stick to the style of the rest of the file.